### PR TITLE
[REF] website_slides: convert question deletion confirmation dialog

### PR DESCRIPTION
**Pull request by flch-odoo**
Convert to owl.

Task-3582733

### CODE REVIEW

**Review by thjo-odoo — APPROVED** [review]
Hello @flch-odoo,
As pointed by AWA and LOMA in this [task](https://www.odoo.com/web#id=3558059&cids=1&menu_id=4720&active_id=3510874&model=project.task&view_type=form) this refactor should probably target master instead of 17.0.
Otherwise it is cleanly done and remove some now useless code.
Cheers!

File: addons/website_slides/static/src/js/slides_course_quiz.js [hunk 1]
```diff
@@ -547,17 +547,27 @@
...
-            var question = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
-            new ConfirmationDialog(this, {
-                questionId: question.data('questionId'),
-                questionTitle: question.data('title')
-            }).open();
+            const question = ev.currentTarget.closest('.o_wslides_js_lesson_quiz_question');
+            const questionId = parseInt(question.dataset.questionId);
+            this.call("dialog", "add", ConfirmationDialog, {
+                title: _t('Delete Question'),
+                body: markup(_t('Are you sure you want to delete this question "<strong>%s</strong>"?', question.dataset.title)),
```
Comment (std-odoo) [inline addons/website_slides/static/src/js/slides_course_quiz.js:561]: 🔥 XSS
image

```html
<img src=x onerror="alert(1)"/>
```

File: addons/website_slides/static/src/js/slides_course_quiz.js [hunk 2]
```diff
@@ -547,17 +547,27 @@
...
+                cancel: () => {
+                },
+                cancelLabel: _t('No'),
+                confirm: async () => {
+                    await this.orm.unlink("slide.question", [questionId]);
```
Comment (std-odoo) [inline addons/website_slides/static/src/js/slides_course_quiz.js:566]: picky std: most of the code use single quote (same above)

**Review by std-odoo — COMMENTED** [review]
@flch-odoo Hi :) 

The refactor look good to me, except for the XSS 🔥

Cheers

**Comment by flch-odoo** [comment]
Thanks for the review, @std-odoo!

**Review by std-odoo — APPROVED** [review]
LGTM, thanks :)

**Comment by std-odoo** [comment]
@tde-banana-odoo Hi :) 

Can I get a ci/security overwrite here ?

The reason is
> We use `markup` to write in bold the question title, and we escape manually `questionTitle`

**Comment by std-odoo** [comment]
@flch-odoo Sorry one last comment, can you escape on the same line as `markup` ? (so if we remove the escape it will trigger a ci/security)

**Review by std-odoo — APPROVED** [review]
LGTM

@tde-banana-odoo Hi :) Can I get a ci/security overwrite ?
(we are using `markup` to show the title in bold, and we escape manually the title)

**Comment by tde-banana-odoo** [comment]
@robodoo override=ci/security

Reason: usage of markup to show the title in bold; title is manually escaped then included in markup

**Comment by std-odoo** [comment]
@robodoo r+

**Pull request closed without merging**